### PR TITLE
Use 409 error code for duplicate issuer when adding km

### PR DESCRIPTION
**Pull request by menakaj**
Fixes https://github.com/wso2-enterprise/choreo/issues/9642

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java [hunk 1]
```diff
@@ -538,8 +538,12 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
                                 APIUtil.getInternalOrganizationDomain(keyManagerConfigurationDTO.getOrganization()));
                 keyManagerConfigurationDTO.setExternalReferenceId(identityProvider.getResourceId());
             } catch (IdentityProviderManagementException e) {
+                ExceptionCodes exceptionCode = ExceptionCodes.IDP_ADDING_FAILED;
+                if (e.getMessage().contains("has already been registered")) {
+                    exceptionCode = ExceptionCodes.DUPLICATE_ISSUER;
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:543]: rather check for exception message can't we check for existence

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java (hunk 1 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java:543]: To check the issuer, we have to get all the idps registered in the org one by one and check the org right? Do we have any other way. 
As I checked in apimgt db we do not keep the issuer in a separate field.

**Review by tharindu1st — DISMISSED** [review]

**Review by malinthaprasan — APPROVED** [review]

**Comment by malinthaprasan** [comment]
Restarted failed jobs. @menakaj pls check the status

**Comment by malinthaprasan** [comment]
restarted build after merging https://github.com/wso2/product-apim/pull/12920

**Comment by menakaj** [comment]
Closing the pr due to the integration test failure. Will reopen once the issue is identified and addressed.

**Pull request closed without merging**